hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3492,6 +3515,28 @@ func (r *HostedClusterReconciler) validateReleaseImage(ctx context.Context, hc *
 	return isValidReleaseVersion(version, currentVersion, hc.Spec.Networking.NetworkType)
 }
 
+func isProgressing(ctx context.Context, hc *hyperv1.HostedCluster) (bool, error) {
+	for _, condition := range hc.Status.Conditions {
+		switch string(condition.Type) {
+		case string(hyperv1.SupportedHostedCluster):
+		case string(hyperv1.ValidHostedClusterConfiguration):
+		case string(hyperv1.ValidReleaseImage):
+		case string(hyperv1.ReconciliationActive):
+			if condition.Status == metav1.ConditionFalse {
```

Comment:
Nitpicky, but why use a switch case with three empty cases that are irrelevant rather than simply an `if` for the one type we actually care about?

Reply:
My suggestion would be to check every condition explicitly
```
condition1 := meta.FindStatusCondition(hcluster.Status.Conditions, string(condition1))
condition2:= meta.FindStatusCondition(hcluster.Status.Conditions, string(condition2))
...
```

That also forces you to handle the empty case (unknown) explicitly, right now with the loop we are implicitly saying no condition is ok.